src/routes/updateProfile.ts

```diff
@@ -25,174 +25,182 @@ export const updateProfile = async (
         !requestBody.profile.nft.collectionAddress ||
-        !("tokenId" in requestBody.profile.nft) ||
+        !('tokenId' in requestBody.profile.nft) ||
         // tokenId could be an empty string, so only perform a typecheck here.
-        typeof requestBody.profile.nft.tokenId !== "string")
-    ) {
-      throw new Error("NFT needs chainId, collectionAddress, and tokenId.");
-    }
-    // Validate chainId supported.
-    if (
-      "nft" in requestBody.profile &&
-      requestBody.profile.nft &&
-      (!("chainId" in requestBody.profile.nft) ||
-        !requestBody.profile.nft.chainId ||
-        !(requestBody.profile.nft.chainId in CHAINS))
+        typeof requestBody.profile.nft.tokenId !== 'string')
     ) {
-      throw new Error(
-        `NFT's chainId must be one of: ${Object.keys(CHAINS).join(", ")}`
-      );
+      throw new Error('NFT needs chainId, collectionAddress, and tokenId.')
     }
   } catch (err) {
-    console.error("Parsing request body", err);
+    console.error('Parsing request body', err)
 
     return respond(400, {
-      error: "Invalid body",
-      message: err instanceof Error ? err.message : `${err}`,
-    });
+      error: err instanceof Error ? err.message : `${err}`,
+    })
   }
 
   // Get existing profile.
-  let existingProfile: Profile = { ...EMPTY_PROFILE };
+  let existingProfile: Profile = { ...EMPTY_PROFILE }
+  let existingProfileId: number | undefined
   try {
-    const stringifiedData = await env.PROFILES.get(getProfileKey(publicKey));
-    if (stringifiedData) {
-      existingProfile = JSON.parse(stringifiedData);
+    const _profile = await getProfileFromPublicKey(env, publicKey)
+    if (_profile) {
+      existingProfile = _profile
+      existingProfileId = _profile.id
     }
   } catch (err) {
-    console.error("Profile retrieval or parsing", err);
+    console.error('Profile retrieval', err)
 
     return respond(500, {
-      error: "Failed to retrieve or parse existing profile",
-      message: err instanceof Error ? err.message : `${err}`,
-    });
+      error:
+        'Failed to retrieve existing profile: ' +
+        (err instanceof Error ? err.message : `${err}`),
+    })
   }
 
   // Validate nonce to prevent replay attacks.
   if (requestBody.profile.nonce !== existingProfile.nonce) {
     return respond(401, {
-      error: "Invalid body",
-      message: `Invalid nonce. Expected: ${existingProfile.nonce}`,
-    });
+      error: `Invalid nonce. Expected: ${existingProfile.nonce}`,
+    })
   }
 
-  const { name, nft } = requestBody.profile;
+  const { name, nft } = requestBody.profile
 
   // If setting name, verify unique.
-  if (typeof name === "string") {
+  if (typeof name === 'string') {
     try {
-      if (await env.PROFILES.get(getPublicKeyForNameTakenKey(name))) {
+      const namedProfile = await getProfileFromName(env, name)
+      // Only error if profile that is not the existing profile has the name.
```

Comment:
????? I half understand this, but I don't understand why this error might occur?

Reply:
this is just a normal name uniqueness check, but we don't complain if the current profile has the name, since they may not be doing a partial update and just include the name and NFT together 🤷🏻‍♂️